src/UIKit/UIContentSizeCategory.cs

```diff
@@ -0,0 +1,38 @@
+﻿#if !WATCH
+
+using System;
+using System.Runtime.InteropServices;
+
+using XamCore.ObjCRuntime;
+using XamCore.Foundation;
+
+namespace XamCore.UIKit {
+	static public partial class UIContentSizeCategoryExtensions {
+
+		[DllImport (Constants.UIKitLibrary)]
+		static extern nint /* NSComparisonResult */ UIContentSizeCategoryCompareToCategory (IntPtr /* NSString */ lhs, IntPtr /* NSString */ rhs);
+
+		[iOS (11, 0), TV (11, 0)]
+		public static NSComparisonResult Compare (UIContentSizeCategory category1, UIContentSizeCategory category2)
+		{
+			if (!UIContentSizeCategory.IsDefined (typeof (UIContentSizeCategory), category1))
+				throw new ArgumentException ($"Unknown 'UIContentSizeCategory' value", nameof (category1));
+
+			if (!UIContentSizeCategory.IsDefined (typeof (UIContentSizeCategory), category2))
+				throw new ArgumentException ($"Unknown 'UIContentSizeCategory' value", nameof (category2));
+
+			return (NSComparisonResult)(long)UIContentSizeCategoryCompareToCategory (category1.GetConstant ().Handle, category2.GetConstant ().Handle);
```

Comment:
`GetConstant` can return `null` if the enum value does not exists on the OS version being used. This will lead to a `NullReferenceException` and it's very likely to work with ObjC.

If you change `.Handle` to `.GetHandle ()` you'll use an extension method that deals with `null` (and return `IntPtr.Zero`).

Reply:
I guess in this case it's fine given that we do a `UIContentSizeCategory.IsDefined` but it's not fine for `UIContentSizeCategoryIsAccessibilityCategory`, I'll fix it anyway.